// @docusaurus/plugin-content-blog/lib/blogFrontMatter.js
"use strict";
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateBlogPostFrontMatter = void 0;
const utils_validation_1 = require("@docusaurus/utils-validation");
// NOTE: we don't add any default value on purpose here
// We don't want default values to magically appear in doc metadatas and props
// While the user did not provide those values explicitly
// We use default values in code instead
const BlogTagSchema = utils_validation_1.JoiFrontMatter.alternatives().try(utils_validation_1.JoiFrontMatter.string().required(), utils_validation_1.JoiFrontMatter.object({
    label: utils_validation_1.JoiFrontMatter.string().required(),
    permalink: utils_validation_1.JoiFrontMatter.string().required(),
}));
const BlogFrontMatterSchema = utils_validation_1.JoiFrontMatter.object({
    id: utils_validation_1.JoiFrontMatter.string(),
    title: utils_validation_1.JoiFrontMatter.string().allow(''),
    description: utils_validation_1.JoiFrontMatter.string().allow(''),
    tags: utils_validation_1.JoiFrontMatter.array().items(BlogTagSchema),
    slug: utils_validation_1.JoiFrontMatter.string(),
    draft: utils_validation_1.JoiFrontMatter.boolean(),
    date: utils_validation_1.JoiFrontMatter.string().allow(''), // TODO validate the date better!
}).unknown();
function validateBlogPostFrontMatter(frontMatter) {
    return utils_validation_1.validateFrontMatter(frontMatter, BlogFrontMatterSchema);
}
exports.validateBlogPostFrontMatter = validateBlogPostFrontMatter;
